stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/ui/common/feature/messages/list/MessageListController.kt

```diff
@@ -617,16 +616,13 @@ public class MessageListController(
         val last = newState.messageItems.filterIsInstance<MessageItemState>().lastOrNull()?.stringify()
         logger.d { "[updateMessageList] #messageList; first: $first, last: $last" }
 
-        val newLastMessage =
-            newState.messageItems.lastOrNull { it is MessageItemState || it is SystemMessageItemState }?.let {
-                when (it) {
-                    is MessageItemState -> it.message
-                    is SystemMessageItemState -> it.message
-                    else -> null
-                }
-            }
+        val oldLastMessage = _messageListState.value.lastItemOrNull<HasMessageListItemState>()?.message
+        val newLastMessage = newState.lastItemOrNull<HasMessageListItemState>()?.message
 
-        val newMessageState = getNewMessageState(newLastMessage, lastLoadedMessage)
+        val newMessageState = when (oldLastMessage?.id != newLastMessage?.id) {
+            true -> getNewMessageState(newLastMessage, lastLoadedMessage)
+            else -> null
+        }
```

Comment:
The actual fix is here.

We have to drop `newMessageState` to `null` if message list did not change.